### PR TITLE
Feature/remove both lib server folders

### LINKED ISSUE #936: Restructuring the project
Using the Meteor Guide, and previous project experience/design decisions, propose a common module structure for the Apinf project. Reorganize code to follow the decided-on structure.

### LINKED ISSUE #1556: Deprecate /both, /lib, and /server folders
There are currently three generic folders in the root level of our project:

- `/both`
- `/lib`
- `/server`

The above folders are left from our original project structure. Move files from the generic folders into relevant locations in our project, and delete the empty folders.

### LINKED ISSUE #1557: Add missing i18n strings to import API Settings/Documentation
The API Settings/Documentation import modules have missing i18n tokens. Add tokens where relevant, and move the English translation strings into en.i18n.js

### LINKED ISSUE #1558: Deprecate FlashMessages package
We are currently using two modules to display messages to users:

- FlashMessages
- sAlert

Deprecate the FlashMessages module in favor of sAlert

### LINKED ISSUE #1559: Enable internationalization for sAlert prompts
There are a few sAlert strings in our UI that are not internationalized. Find sAlert prompts that lack i18n tokens, add the tokens, and move the UI strings into the `en.i18n.js`.

### LINKED ISSUE #1575: Remove editApi component
Since refactoring the Api module, we no longer use the `editApi` sub-component. Remove the `editApi` sub-component including routing and i18n.

### LINKED ISSUE #1577: Replace i18n hyphens with underscores
We decided, at one point, to use underscores in our i18n keys. This may have been due to a limitation of the TAPi18n library at the time. In any case, we are mixing hyphens and underscores in our i18n keys.

Rename all i18n keys to contain underscores instead of hyphens, to the extent possible.

### LINKED ISSUE #1578: Remove `manageApis` component
After refactoring the Apis module, and with recent Catalogue re-design, the `manageApis` sub-component is no longer used. Remove the `manageApis` sub-component, route, and i18n.

### LINKED ISSUE #1579: Update Settings schema i18n strings
The Settings schema i18n strings do not reflect the current Settings schema structure. Remove unused i18n strings and update based on existing Settings schema.

### PR DESCRIPTION
**Pull request by brylie**
Closes #936 
Closes #1556 
Closes #1557
Closes #1558
Closes #1559 
Closes #1575 
Closes #1577
Closes #1578
Closes #1579

# Changes
- Make final changes to modularize project package structure
  - move files from `/both` folder into relevant locations
  - move files from `/lib` folder into relevant locations
  - move files from `/server` folder into relevant locations
- Add missing i18n tokens where found in cleanup process above
- Adjust import/exports
- Deprecate FlashMessages package
- Add missing sAlert i18n functionality
- Automatic lint
- remove `editApi` component
- remove `manageApis` component

### CODE REVIEW

**Comment by brylie** [comment]
@apinf/core please review ASAP

**Comment by bajiat** [comment]
@jykae Could you review this?

**Comment by bajiat** [comment]
@marla-singer Could you review this PR?
@jykae Asking another developer since there was no confirmation from you

File: settings/server/oauth_configuration.js [hunk 1]
```diff
@@ -1,11 +1,13 @@
 if (Meteor.isServer) {
```
Comment (marla-singer) [inline settings/server/oauth_configuration.js]: @brylie File is in the folder "Server", I think there is no need in this line

**Comment by marla-singer** [comment]
@brylie I can't test the site feature because login and sign Up don't work.
Server console error is
`Exception while invoking method 'ATCreateUserServer' Error: After filtering out keys not in the schema, your object is now empty`

**Comment by marla-singer** [comment]
@brylie don't work thefeatures associated with the user schema: sign up, sign in, sign out, change password. 
Server console error one of this:
`Exception while invoking method 'ATCreateUserServer'
Error: After filtering out keys not in the schema, your object is now empty`

`Exception while invoking method 'logout' Error: After filtering out keys not in the schema, your modifier is now empty
`

`Exception while invoking method 'changePassword' Error: services.password.reset is not allowed by the schema`

**Comment by brylie** [comment]
Hm, do you have any ideas what might be broken here?

**Comment by brylie** [comment]
Perhaps there is an import issue related to the schema.js file.

**Comment by brylie** [comment]
@marla-singer I have found an error in my *schemas.js* file. Please review.

**Comment by marla-singer** [comment]
@brylie Works. 
I didn't find another problems. Could I merge it or it should be reviewed another developer too?

**Comment by brylie** [comment]
@marla-singer thanks for the quick review.

I have made small changes to 18 files, related to `sAlert` with translation strings. Please review and merge, if there are no other issues.

File: ratings/client/rating.js [hunk 2]
```diff
@@ -60,10 +60,10 @@ Template.apiBackendRating.events({
     // Make sure there is a Meteor user ID for voting
     if (Meteor.userId() === null) {
       // Get translated user message
-      const userMessage = TAPi18n.__('api-backend-rating-anonymous');
+      const message = TAPi18n.__('api-backend-rating-anonymous');
```
Comment (marla-singer) [inline ratings/client/rating.js]: Preferably use underscore insted of dash

File: users/client/profile/profile.js [hunk 3]
```diff
@@ -20,8 +23,11 @@ Template.profile.rendered = function () {
   if (currentUser) {
     // Ask user to set username if it is not set.
     if (!currentUser.username) {
-      const setUsernameMsg = TAPi18n.__('profile-setUsername');
-      sAlert.info(setUsernameMsg);
+      // Get username 'update needed' message
+      const message = TAPi18n.__('profile-setUsername');
```
Comment (marla-singer) [inline users/client/profile/profile.js]: Preferably use underscore insted of dash

File: documentation/client/manage/manage.js [hunk 4]
```diff
@@ -13,7 +13,7 @@ Template.manageApiDocumentationModal.onCreated(function () {
         placement: 'left',
       },
     },
-    'uploadApiDocumentation': {
+    'uploadApiDocumentation': {TAPi18n.__('manageApiDocumentationModal_DeletedFile_ConfirmationMessage')
```
Comment (marla-singer) [inline documentation/client/manage/manage.js]: TAPi18n is needless in this line?

File: users/client/profile/profile.js (hunk 3 above)
Comment (brylie) [inline users/client/profile/profile.js]: Good catch.

File: documentation/client/manage/manage.js (hunk 4 above)
Comment (brylie) [inline documentation/client/manage/manage.js]: Hm, I am not sure. I will look at that.

Comment (brylie) [inline documentation/client/manage/manage.js]: Jep. It seemed out of place.

**Comment by brylie** [comment]
I am going to double-check our i18n hyphens.

File: ratings/client/rating.js [hunk 5]
```diff
@@ -60,7 +60,7 @@ Template.apiBackendRating.events({
     // Make sure there is a Meteor user ID for voting
     if (Meteor.userId() === null) {
       // Get translated user message
-      const message = TAPi18n.__('api-backend-rating-anonymous');
+      const message = TAPi18n.__('apiBackendRating-anonymousError');
```
Comment (marla-singer) [inline ratings/client/rating.js]: One of these is here

**Comment by brylie** [comment]
OK, I was looking at i18n strings and realized we are no longer using the `editApi` component. So, I removed the component entirely.

File: ratings/client/rating.js (hunk 2 above)
Comment (brylie) [inline ratings/client/rating.js]: This has been changed.

**Comment by marla-singer** [comment]
Fix this label 
https://github.com/apinf/platform/blob/develop/apis/client/view/settings/settings.html#L9

**Comment by brylie** [comment]
Made the requested fixes. Please review.

**Comment by brylie** [comment]
@marla-singer I have fixed the conflicts. Please review and merge, if possible, ASAP.

**Pull request merged**